noir/noir-repo/aztec_macros/src/transforms/note_interface.rs

```diff
@@ -0,0 +1,510 @@
+    Ok(struct_ast.types.remove(0))
+}
+
+fn generate_note_deserialize_content(
+    note_type: &str,
+    note_fields: &[(String, String)],
+    note_serialize_len: &String,
+    impl_span: Option<Span>,
+) -> Result<NoirFunction, AztecMacroError> {
+    let function_source =
+        generate_note_deserialize_content_source(note_type, note_fields, note_serialize_len);
+
+    let (function_ast, errors) = parse_program(&function_source);
+    if !errors.is_empty() {
+        return Err(AztecMacroError::CouldNotImplementNoteSerialization {
+            secondary_message: Some("Failed to parse Noir macro code (fn deserialize_content). This is either a bug in the compiler or the Noir macro code".to_string()),
+            span: impl_span
+        });
+    }
+
+    let mut function_ast = function_ast.into_sorted();
+    let mut noir_fn = function_ast.functions.remove(0);
+    noir_fn.def.visibility = ItemVisibility::Public;
+    Ok(noir_fn)
+}
+
+fn generate_note_serialize_content(
+    note_type: &str,
+    note_fields: &[(String, String)],
+    note_serialize_len: &String,
+    impl_span: Option<Span>,
+) -> Result<NoirFunction, AztecMacroError> {
+    let function_source =
+        generate_note_serialize_content_source(note_type, note_fields, note_serialize_len);
+
+    let (function_ast, errors) = parse_program(&function_source);
+    if !errors.is_empty() {
+        return Err(AztecMacroError::CouldNotImplementNoteSerialization {
+            secondary_message: Some("Failed to parse Noir macro code (fn seserialize_content). This is either a bug in the compiler or the Noir macro code".to_string()),
+            span: impl_span
+        });
+    }
+
+    let mut function_ast = function_ast.into_sorted();
+    let mut noir_fn = function_ast.functions.remove(0);
+    noir_fn.def.visibility = ItemVisibility::Public;
+    Ok(noir_fn)
+}
+
+fn generate_note_fields_fn(
+    note_type: &str,
+    note_fields: &[(String, String)],
+    impl_span: Option<Span>,
+) -> Result<NoirFunction, AztecMacroError> {
+    let function_source = generate_note_fields_fn_source(note_type, note_fields);
+    let (function_ast, errors) = parse_program(&function_source);
+    if !errors.is_empty() {
+        return Err(AztecMacroError::CouldNotImplementNoteSerialization {
+            secondary_message: Some("Failed to parse Noir macro code (fn fields). This is either a bug in the compiler or the Noir macro code".to_string()),
+            span: impl_span
+        });
+    }
+    let mut function_ast = function_ast.into_sorted();
+    let mut noir_fn = function_ast.functions.remove(0);
+    noir_fn.def.visibility = ItemVisibility::Public;
+    Ok(noir_fn)
+}
+
+fn generate_compute_note_content_hash(
+    note_type: &String,
+    impl_span: Option<Span>,
+) -> Result<NoirFunction, AztecMacroError> {
+    let function_source = format!(
+        "
+        fn compute_note_content_hash(self: {}) -> Field {{
+            // TODO(#1205) Should use a non-zero generator index.
```

Comment:
Surely we don't need to include the comment? :joy: 
Let's move that TODO comment and issue reference to this code instead

Reply:
Kept it so a generic search of TODO(#1205) would yield the result in the codebase, but I know it's weird 🤣